### PR TITLE
Dockerize Nexus

### PR DESCRIPTION
**Pull request by tuky191**
# Dockerize Nexus

  - Uses docker and docker compose to build up the infrastructure
  - Builds SUI devnet with 4 validators, 1 fullnode and a faucet
  - Deploys the Nexus package
  - Deploys or starts Ollama locally
  - Bootstraps the llama model, airdrops gas from faucet and saves
 ```   
  node_details = {
        "node_id": node_id,
        "llama_id": llama_id,
        "llama_owner_cap_id": llama_owner_cap_id,
    }
```
 - Creates tools, events and example containers
  

## Summary by CodeRabbit

- **New Features**
	- Introduced a new environment configuration for Docker, specifying essential variables for application operation.
	- Added a new script for publishing packages using the Sui client.
	- New Docker Compose configurations for Sui network setup with multiple services, including validators and full nodes.
	- Enhanced example scripts with dynamic command-line argument parsing and improved error handling.
	- Implemented a new utility function for creating a Sui client with an airdrop feature.
	- Added a new configuration file for a Sui full node setup, detailing operational parameters.
	- Introduced a new configuration for validator accounts and parameters in the genesis setup.
	- Added support for specifying tool URLs in event handling for improved flexibility.
	- Enhanced Docker configurations to streamline service management and improve build efficiency.

- **Bug Fixes**
	- Improved error handling in various scripts to provide clearer messages and prevent failures.

- **Documentation**
	- Updated README files with clearer instructions and broader OS support.
	- Added new sections in documentation for Docker and Events.

- **Chores**
	- Modified `.gitignore` and `.dockerignore` files to refine file management.
	- Updated workflow configurations to trigger on relevant file changes.

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# Force LF line endings for all files
+* text=auto eol=lf
```
Comment (pb-at-talus) [inline .gitattributes:2]: windows ey? :kiwi_fruit:

File: docker/README.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# README
+
+docker context use default
```
Comment (pb-at-talus) [inline docker/README.md:3]: TODO

**Review by pb-at-talus — COMMENTED** [review]
Thanks! Looks great, will try to run it later

File: docker/README.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  - 4 validators
+  - faucet
+  - fullnode
+- nexus
+  - events
+  - tools
+  - examples
+- ollama (only on windows and linux)
+
+There are also couple startup services
```
Comment (loverdos) [inline docker/README.md:17]:
```suggestion
There are also a few startup services
```

File: docker/nexus/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,43 @@
+FROM python:3.10-slim AS builder
```
Comment (loverdos) [inline docker/nexus/Dockerfile:1]: using syntax is considered good practice.

```suggestion
#syntax=docker/dockerfile:1

FROM python:3.10-slim AS builder
```

File: docker/nexus/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+COPY . .
+
+RUN if [ "$INSTALL_RUST" = "true" ]; then \
+        apt-get update && apt-get install -y --no-install-recommends curl build-essential && \ 
+        curl https://sh.rustup.rs -sSf | sh -s -- -y && \
+        . $HOME/.cargo/env && \ 
+        rustup update && \
+        rustup default stable; \
+    fi
```
Comment (loverdos) [inline docker/nexus/Dockerfile:16]: We can make it look like a shell script.

```suggestion
RUN <<EOF
  set -eo pipefail
  
  if [ "$INSTALL_RUST" = "true" ]; then
     apt-get update
     apt-get install -y --no-install-recommends curl build-essential
     curl https://sh.rustup.rs -sSf | sh -s -- -y
     . $HOME/.cargo/env
     rustup update
     rustup default stable
  fi
EOF
```

File: docker/nexus/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+RUN pip install uv && \
```
Comment (loverdos) [inline docker/nexus/Dockerfile:18]: Same as the previous RUN

**Comment by pb-at-talus** [comment]
`just example ig-post-planner` fails if ollama is installed on host machine already with a port conflict

**Review by pb-at-talus — DISMISSED** [review]
`just example ig-post-planner` fails if ollama is installed on host machine already with a port conflict.

Possibly people could have it already installed. Perhaps we can bind to a non default port?

Other than this issue, everything works on my end, thanks!

**Pull request merged**